### PR TITLE
Flatpak support

### PR DESCRIPTION
**Pull request by bscubed**
Adds Flatpak build capabilities to Travis-CI, bringing cross-platform support to all Linux distributions.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4383)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -61,6 +61,16 @@ matrix:
       script: "./.travis/linux-mingw/build.sh"
       after_success: "./.travis/linux-mingw/upload.sh"
       cache: ccache
+    - if: repo =~ ^.*\/(citra-canary|citra-nightly)$
```
Comment (j-selby) [inline .travis.yml]: Is there a reason we aren't testing this on the main repo, and just not deploying? i.e if a PR generates a regression in flatpak, that won't get caught until canary/merge.

File: .travis/linux-flatpak/docker.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+chmod -R 600 "$HOME/.ssh"
+chown -R root "$HOME/.ssh"
+chmod 600 "$SSH_KEY"
+ssh-add "$SSH_KEY"
+echo "[$FLATPAK_SSH_HOSTNAME]:$FLATPAK_SSH_PORT,[$(dig +short $FLATPAK_SSH_HOSTNAME)]:$FLATPAK_SSH_PORT $SSH_PUBLIC_KEY" > ~/.ssh/known_hosts
+
+# Configure GPG keys
+gpg2 --import "$GPG_KEY"
+
+# Download the Citra compatibility list
```
Comment (j-selby) [inline .travis/linux-flatpak/docker.sh]: Might want to note that this was a workaround for flatpak's builder.

Comment (bscubed) [reply .travis/linux-flatpak/docker.sh]: In reference to the compatibility list?

File: .travis.yml (hunk 1 above)
Comment (bscubed) [reply .travis.yml]: Ultimately, we're only gonna be deploying on Nightly/Canary anyways, and there's only really one major point of conflict in the build, so as long as nothing changes one specific line of code in the base CMakeList.txt everything should be fine.

File: .travis/linux-flatpak/docker.sh (hunk 2 above)
Comment (j-selby) [reply .travis/linux-flatpak/docker.sh]: yep

**Pull request merged**